src/material/datepicker/calendar.ts

```diff
@@ -149,8 +151,40 @@ export class MatCalendarHeader<D> {
       return this._dateAdapter.getYear(date1) == this._dateAdapter.getYear(date2);
     }
     // Otherwise we are in 'multi-year' view.
-    return Math.floor(this._dateAdapter.getYear(date1) / yearsPerPage) ==
-        Math.floor(this._dateAdapter.getYear(date2) / yearsPerPage);
+    return this._isSameMultiYearView(date1, date2);
+  }
+
+  private _isSameMultiYearView(date1: D, date2: D): boolean {
```

Comment:
These functions are still duplicated; it would be better to extract them to one place where they accept a `dateAdapter` as an argument rather than defining them in both classes.